app/services/solidus_bolt/payments/credit_sync_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module Payments
+    class CreditSyncService < BaseService
+      attr_reader :payment, :amount, :response_code, :refund_reason, :order
+
+      def initialize(payment:, amount:, response_code:)
+        @payment = payment
+        @amount = amount
+        @refund_reason = Spree::RefundReason.return_processing_reason&.id
+        @order = payment.order
+        super
+      end
+
+      def call
+        Spree::Refund.create(
+          payment: payment,
+          amount: amount.to_f / 100,
+          transaction_id: response_code,
+          refund_reason_id: refund_reason
+        )
```

Comment:
Shouldn't we check if a `Spree::Refund` with the `transaction_id` already exists or not?

If a `Spree::Refund` with the given `transaction_id` exists then we should let the process fail silently.

Reply:
can't we issue multiple partial refunds to the same payment?

Reply:
fixed 4317db294a8588c40f43ff0e66d4f8e4285465f2